webpack/components/TypeAhead/pf4Search/TypeAheadInput.js

```diff
@@ -0,0 +1,31 @@
+import React, { useRef } from 'react';
+import { TextInput } from '@patternfly/react-core';
+
+import useEventListener from '../../../utils/useEventListener';
+import { commonInputPropTypes } from '../helpers/commonPropTypes';
+
+const TypeAheadInput = ({ onKeyPress, onInputFocus, passedProps }) => {
+  const inputRef = useRef(null);
+  const { onChange, ...downshiftProps } = passedProps;
+
+  // What patternfly4 expects for args and what downshift creates as a function is different,
+  // downshift only expects the event handler
+  const onChangeWrapper = (_userValue, event) => onChange(event);
+
+  useEventListener('keydown', onKeyPress, inputRef.current);
+
+  return (
+    <TextInput
+      {...downshiftProps}
+      ref={inputRef}
```

Comment:
:+1: for switching to the useRef and non-function form of the prop.  I was also going to suggest dropping `ref` in favor of  the `innerRef` prop provided by PF4, but both `ref` and `innerRef` seem to put the event listener on the `<input>` element and both work fine.  So I guess either is fine.